tools/swarm/src/compose.rs

```diff
@@ -316,23 +319,23 @@ struct CompactPeerEnv {
     chain_id: ChainId,
     key_pair: KeyPair,
     genesis_public_key: PublicKey,
-    /// Genesis private key is only needed for a peer that is submitting the genesis block
-    genesis_private_key: Option<PrivateKey>,
+    /// Genesis file is only needed for a peer that is submitting the genesis block
+    genesis_file: Option<String>,
     p2p_addr: SocketAddr,
     api_addr: SocketAddr,
     trusted_peers: BTreeSet<PeerId>,
 }
 
 impl From<CompactPeerEnv> for FullPeerEnv {
     fn from(value: CompactPeerEnv) -> Self {
-        let (genesis_private_key_algorithm, genesis_private_key_payload, genesis_file) = value
-            .genesis_private_key
-            .map_or((None, None, None), |private_key| {
-                let (algorithm, payload) = private_key.to_bytes();
+        let (genesis_file, genesis_signature) =
+            value.genesis_file.map_or((None, None), |genesis_path| {
                 (
-                    Some(algorithm),
-                    Some(payload),
-                    Some(PATH_TO_GENESIS.to_string()),
+                    Some(genesis_path),
+                    Some(generate_hex_string_signature(
```

Comment:
Wait, why you generate a signature here?

`CompactPeerEnv` & `FullPeerEnv` are emptyheaded structs purely for data representation purposes. They don't do any such logic. `genesis_signature` should be provided to them from outside.